SoftLayer/CLI/firewall/detail.py

```diff
@@ -12,31 +12,67 @@
 
     firewall_type, firewall_id = firewall.parse_id(identifier)
-    _firewall = mgr.get_instance(firewall_id)
 
-    table = formatting.KeyValueTable(['name', 'value'])
-    table.align['name'] = 'r'
-    table.align['value'] = 'l'
+    if firewall_type == 'vs' or firewall_type == 'server' or firewall_type == 'vlan' or firewall_type == 'multiVlan':
 
-    table.add_row(['id', _firewall.get('id')])
-    table.add_row(['primaryIpAddress', _firewall.get('primaryIpAddress')])
-    table.add_row(['datacenter', utils.lookup(_firewall, 'datacenter', 'longName')])
-    table.add_row(['networkVlan', utils.lookup(_firewall, 'networkVlan', 'name')])
-    table.add_row(['networkVlaniD', utils.lookup(_firewall, 'networkVlan', 'id')])
+        if firewall_type == 'vlan':
+            _firewall = mgr.get_instance(firewall_id)
+
+            table = formatting.KeyValueTable(['name', 'value'])
+            table.align['name'] = 'r'
+            table.align['value'] = 'l'
+
+            table.add_row(['id', _firewall.get('id')])
+            table.add_row(['primaryIpAddress', _firewall.get('primaryIpAddress')])
+            table.add_row(['datacenter', utils.lookup(_firewall, 'datacenter', 'longName')])
+            table.add_row(['networkVlan', utils.lookup(_firewall, 'networkVlan', 'name')])
+            table.add_row(['networkVlaniD', utils.lookup(_firewall, 'networkVlan', 'id')])
+
+            rules = mgr.get_dedicated_fwl_rules(firewall_id)
+            table.add_row(['rules', get_rules_table(rules)])
+
+        if firewall_type == 'multiVlan':
+            _firewall = mgr.get_instance(firewall_id)
+
+            table = formatting.KeyValueTable(['name', 'value'])
+            table.align['name'] = 'r'
+            table.align['value'] = 'l'
+
+            table.add_row(['name', utils.lookup(_firewall, 'networkGateway', 'name')])
+            table.add_row(['datacenter', utils.lookup(_firewall, 'datacenter', 'longName')])
+            table.add_row(['public ip', utils.lookup(_firewall, 'networkGateway', 'publicIpAddress', 'ipAddress')])
+            table.add_row(['private ip', utils.lookup(_firewall, 'networkGateway', 'privateIpAddress', 'ipAddress')])
+            table.add_row(['public  ipv6', utils.lookup(_firewall, 'networkGateway', 'publicIpv6Address', 'ipAddress')])
+            table.add_row(['public vlan', utils.lookup(_firewall, 'networkGateway', 'publicVlan', 'vlanNumber')])
+            table.add_row(['private vlan', utils.lookup(_firewall, 'networkGateway', 'privateVlan', 'vlanNumber')])
+            table.add_row(['type', _firewall.get('firewallType')])
+
+            if credentials:
+                table.add_row(['FortiGate username', utils.lookup(_firewall, 'managementCredentials', 'username')])
+                table.add_row(['FortiGate password', utils.lookup(_firewall, 'managementCredentials', 'password')])
+
+            rules = mgr.get_dedicated_fwl_rules(firewall_id)
+            table.add_row(['rules', get_rules_table(rules)])
```

Comment:
please add a conditional to rules is empty

Reply:
A conditional to print rules was added where rules variable is empty